modules/graph-data-science/pages/node-classification.adoc

```diff
@@ -0,0 +1,481 @@
+    }
+  }, {
+    APPEARS_WITH_UNDIRECTED: {
+      type: 'APPEARED_WITH',
+      orientation: 'UNDIRECTED',
+      aggregation: 'SINGLE',
+      properties: ['times']
+    },
+    APPEARS_WITH_DIRECTED: {
+      type: 'APPEARED_WITH',
+      orientation: 'NATURAL',
+      properties: ['times'],
+      aggregation: 'SINGLE'
+    },
+    ALLY_UNDIRECTED: {
+      type: 'ALLY',
+      orientation: 'UNDIRECTED',
+      aggregation: 'SINGLE'
+    },
+    ALLY_DIRECTED: {
+      type: 'ALLY',
+      orientation: 'NATURAL',
+      aggregation: 'SINGLE'
+    },    
+    ENEMY_UNDIRECTED: {
+      type: 'ENEMY',
+      orientation: 'UNDIRECTED',
+      aggregation: 'SINGLE'
+    },
+    ENEMY_DIRECTED: {
+      type: 'ENEMY',
+      orientation: 'NATURAL',
+      aggregation: 'SINGLE'
+    }
+   
+});
+----
+
+and then we use this graph to calculate the https://neo4j.com/docs/graph-data-science/current/algorithms/page-rank/[PageRank], https://neo4j.com/docs/graph-data-science/current/algorithms/betweenness-centrality/[Betweenness Centrality], and https://neo4j.com/docs/graph-data-science/current/algorithms/hits/[Hyperlink-Induced Topic Search] (HITS) of each node and write those values back to the database:
+
+[source, cypher]
+----
+// pageRank
+CALL gds.pageRank.write('marvel-character-graph',{
+     relationshipTypes: ['APPEARS_WITH_DIRECTED'],
+     writeProperty: 'appeared_with_pageRank'
+});
+CALL gds.pageRank.write('marvel-character-graph',{
+     relationshipTypes: ['ALLY_DIRECTED'],
+     writeProperty: 'ally_pageRank'
+});
+CALL gds.pageRank.write('marvel-character-graph',{
+     relationshipTypes: ['ENEMY_DIRECTED'],
+     writeProperty: 'enemy_pageRank'
+});
+
+// betweenness
+CALL gds.betweenness.write('marvel-character-graph',{
+     relationshipTypes: ['APPEARS_WITH_UNDIRECTED'],
+     writeProperty: 'appeared_with_betweenness'
+});
+CALL gds.betweenness.write('marvel-character-graph',{
+     relationshipTypes: ['ALLY_UNDIRECTED'],
+     writeProperty: 'ally_betweenness'
+});
+CALL gds.betweenness.write('marvel-character-graph',{
+     relationshipTypes: ['ENEMY_UNDIRECTED'],
+     writeProperty: 'enemy_betweenness'
+});
+
+//HITS
+CALL gds.alpha.hits.write('marvel-character-graph',{
+     relationshipTypes: ['APPEARS_WITH_DIRECTED'],
+     hitsIterations: 50,
+     authProperty: 'appeared_with_auth',
+     hubProperty: 'appeared_with_hub'
+});
+CALL gds.alpha.hits.write('marvel-character-graph',{
+     relationshipTypes: ['ALLY_DIRECTED'],
+     hitsIterations: 50,
+     authProperty: 'appeared_with_auth',
+     hubProperty: 'appeared_with_hub'
+});
+CALL gds.alpha.hits.write('marvel-character-graph',{
+     relationshipTypes: ['ENEMY_DIRECTED'],
+     hitsIterations: 50,
+     authProperty: 'appeared_with_auth',
+     hubProperty: 'appeared_with_hub'
+});
+----
+
+We will also want these values added to the in-memory graph for the sake of calculating graph embeddings in the next step, which is achieved through the `.mutate()` command:
```

Comment:
It seems like there should be a nicer way to do this than to run all of the algorithms twice. I think there's a way that you can flush all the values in the in-memory graph back to Neo4j, but I haven't actually tried it.

If that's possible, you could just use the `mutate` algos and then write all the results back in one go.

Reply:
See previous comment re: Alicia's queries.